runtime/src/constants.rs

```diff
@@ -0,0 +1,43 @@
+//! A set of constant values used in substrate runtime.
+
+/// Time.
+pub mod time {
+    use node_primitives::{Moment, BlockNumber};
+
+    /// Since BABE is probabilistic this is the average expected block time that
+    /// we are targetting. Blocks will be produced at a minimum duration defined
+    /// by `SLOT_DURATION`, but some slots will not be allocated to any
+    /// authority and hence no block will be produced. We expect to have this
+    /// block time on average following the defined slot duration and the value
+    /// of `c` configured for BABE (where `1 - c` represents the probability of
+    /// a slot being empty).
+    /// This value is only used indirectly to define the unit constants below
+    /// that are expressed in blocks. The rest of the code should use
+    /// `SLOT_DURATION` instead (like the timestamp module for calculating the
+    /// minimum period).
+    ///
+    /// If using BABE with secondary slots (default) then all of the slots will
+    /// always be assigned, in which case `MILLISECS_PER_BLOCK` and
+    /// `SLOT_DURATION` should have the same value.
+    ///
+    /// <https://research.web3.foundation/en/latest/polkadot/BABE/Babe/#6-practical-results>
+    pub const MILLISECS_PER_BLOCK: Moment = 6000;
+    pub const SECS_PER_BLOCK: Moment = MILLISECS_PER_BLOCK / 1000;
+
+    pub const SLOT_DURATION: Moment = MILLISECS_PER_BLOCK;
+
+    // 1 in 4 blocks (on average, not counting collisions) will be primary BABE blocks.
+    pub const PRIMARY_PROBABILITY: (u64, u64) = (1, 4);
+
+    pub const EPOCH_DURATION_IN_BLOCKS: BlockNumber = 30; // 10 * MINUTES;
```

Comment:
Is the 10*MINUTES comment still relevant?

Reply:
Good catch, will remove